--- ace_editor_orig/ace_editor/ace_editor.module	2025-07-13 14:09:59
+++ ace_editor/ace_editor.module	2025-08-04 17:28:09
@@ -40,7 +40,7 @@
     '/libraries/ace',
     '/libraries/ace-builds',
     '/' . \Drupal::service('extension.list.module')->getPath('ace_editor') . '/libraries',
-    '/' . \Drupal::service('extension.list.profile')->getPath(\Drupal::installProfile()) . '/libraries/ace',
+    '/' . \Drupal::service('extension.list.module')->getPath('dxpr_builder') . '/vendor/managed/ace-builds/src-min-noconflict',
   ];
 
   foreach ($paths_to_check as $path) {
